projects/aca-content/viewer/src/lib/services/viewer.service.ts

```diff
@@ -0,0 +1,266 @@
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * from Hyland Software. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { ObjectUtils, UserPreferencesService } from '@alfresco/adf-core';
+import { SearchRequest } from '@alfresco/js-api';
+import { Injectable } from '@angular/core';
+import { ContentApiService } from '@alfresco/aca-shared';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ViewerService {
+  constructor(private preferences: UserPreferencesService, private contentApi: ContentApiService) {}
+
+  recentFileFilters = [
+    'TYPE:"content"',
+    '-PATH:"//cm:wiki/*"',
+    '-TYPE:"app:filelink"',
+    '-TYPE:"fm:post"',
+    '-TYPE:"cm:thumbnail"',
+    '-TYPE:"cm:failedThumbnail"',
+    '-TYPE:"cm:rating"',
+    '-TYPE:"dl:dataList"',
+    '-TYPE:"dl:todoList"',
+    '-TYPE:"dl:issue"',
+    '-TYPE:"dl:contact"',
+    '-TYPE:"dl:eventAgenda"',
+    '-TYPE:"dl:event"',
+    '-TYPE:"dl:task"',
+    '-TYPE:"dl:simpletask"',
+    '-TYPE:"dl:meetingAgenda"',
+    '-TYPE:"dl:location"',
+    '-TYPE:"fm:topic"',
+    '-TYPE:"fm:post"',
+    '-TYPE:"ia:calendarEvent"',
+    '-TYPE:"lnk:link"'
+  ];
+
+  /**
+   * Retrieves nearest node information for the given node and folder.
+   *
+   * @param nodeId Unique identifier of the document node
+   * @param folderId Unique identifier of the containing folder node.
+   */
+  async getNearestNodes(nodeId: string, folderId: string, source: string): Promise<{ left: string; right: string }> {
+    const empty = {
+      left: null,
+      right: null
+    };
+
+    if (nodeId && folderId) {
+      try {
+        const ids = await this.getFileIds(source, folderId);
+        const idx = ids.indexOf(nodeId);
+
+        if (idx >= 0) {
+          return {
+            left: ids[idx - 1] || null,
+            right: ids[idx + 1] || null
+          };
+        } else {
+          return empty;
+        }
+      } catch {
+        return empty;
+      }
+    } else {
+      return empty;
+    }
+  }
+
+  /**
+   * Retrieves a list of node identifiers for the folder and data source.
+   *
+   * @param folderId Optional parameter containing folder node identifier for 'personal-files' and 'libraries' sources.
+   */
+  async getFileIds(source: string, folderId?: string): Promise<string[]> {
+    const isClient = this.preferences.get(`${source}.sorting.mode`) === 'client';
+    const [sortKey, sortDirection, previousSortKey, previousSortDir] = this.getSortKeyDir(source);
+
+    if (source === 'personal-files' && folderId) {
+      const orderBy = isClient ? null : ['isFolder desc', `${sortKey} ${sortDirection}`];
+      const nodes = await this.contentApi
+        .getNodeChildren(folderId, {
+          orderBy: orderBy,
+          fields: this.getFields(sortKey, previousSortKey),
+          where: '(isFile=true)'
+        })
+        .toPromise();
+
+      const entries = nodes.list.entries.map((obj) => obj.entry);
```

Comment:
This part: 
```ts
const entries = nodes.list.entries.map((obj) => obj.entry);
      if (isClient) {
        if (previousSortKey) {
          this.sort(entries, previousSortKey, previousSortDir);
        }
        this.sort(entries, sortKey, sortDirection);
      }
      return entries.map((entry) => entry.id);
```

looks similarly for every 'if'. Can you remove duplications? They are multiple ways to achieve that, for example it can be moved after all 'ifs' at the end of function so will be executed for every case